Tests/AbstractWebApplicationTest.php

```diff
@@ -1444,6 +1444,38 @@ public function testDetectRequestUri(
 		);
 	}
 
+	/**
+	 * @testdox  Tests that the application correcty detects the request URI based on the injected data
+	 *
+	 * @param   string|null  $https        Value for $_SERVER['HTTPS'] or null to not set it
+	 * @param   string       $phpSelf      Value for $_SERVER['PHP_SELF']
+	 * @param   string       $requestUri   Value for $_SERVER['REQUEST_URI']
+	 * @param   string       $httpHost     Value for $_SERVER['HTTP_HOST']
+	 * @param   string       $scriptName   Value for $_SERVER['SCRIPT_NAME']
+	 * @param   string       $queryString  Value for $_SERVER['QUERY_STRING']
+	 * @param   string       $expects      Expected full URI string
+	 *
+	 * @covers  \Joomla\Application\AbstractWebApplication
+	 *
+	 * @backupGlobals enabled
+	 * @expectedException \InvalidArgumentException
+	 */
+	public function testDetectRequestUriException()
+	{
+		$mockInput = new Input([]);
+
+		$_SERVER['PHP_SELF'] = 'somthing/framework.php';
+		$_SERVER['HTTP_HOST'] = '';
+		$_SERVER['SCRIPT_NAME'] = 'somthing/framework.php';
+
+		$object = $this->getMockForAbstractClass(AbstractWebApplication::class, [$mockInput]);
+
+		$this->assertSame(
+			$expects,
```

Comment:
@wilsonge The test function has an error here. There is also prams in your doc comment that is not used.